src/ATokenNaive.sol

```diff
@@ -198,15 +203,24 @@ contract ATokenNaive is AToken {
     uint256 amount,
     uint256 index
   ) internal returns (bool) {
-    bool _returnVar = _mintScaled(caller, onBehalfOf, amount, index);
-
     // We increment by `amount` instead of any computed/rebased amounts because
     // `amount` is what actually gets transferred of the underlying asset. We
-    // need our checkpoints to still match up with the underlying asset balance.
-    _writeCheckpoint(_checkpoints[onBehalfOf], _additionFn, amount);
-    _writeCheckpoint(_totalDepositCheckpoints, _additionFn, amount);
+    // need our checkpoints to still match up with underlying asset transactions.
+    Checkpoints.History storage _depositHistory = _depositCheckpoints[onBehalfOf];
+    Checkpoints.push(_depositHistory, Checkpoints.latest(_depositHistory) + amount);
+    Checkpoints.push(
+      _totalDepositCheckpoints, Checkpoints.latest(_totalDepositCheckpoints) + amount
+    );
+
+    return _mintScaled(caller, onBehalfOf, amount, index);
+  }
+
+  function getPastDeposits(address _voter, uint256 _blockNumber) public returns (uint256) {
+    return Checkpoints.getAtBlock(_depositCheckpoints[_voter], _blockNumber);
```

Comment:
And this would become
```suggestion
    return _depositCheckpoints[_voter].getAtBlock(_blockNumber);
```

etc.

there's no other tradeoffs aside readability/preference, so will defer to you, but this is analgous to what we did with math libs of `x.mulWadUp(y)`. Maybe see what syntax Aave uses in their codebase and go with that?